manifests/init.pp

```diff
@@ -1,25 +1,46 @@
-# Default strongSwan class.
+# @summary Installs and manages strongSwan
+#
+# @param package_name
+#   The strongswan package name.
+# @param package_ensure
+#    The desired package state. Can be used to install `latest` or a specific version etc.
+# @param service_name
+#    The strongswan service name.
+# @param service_ensure
+#    The desired `ensure` state for the service.
+# @param service_enable
+#    The desited `enable` state for the service.
+# @param ipsec_d_dir
+#    The directory for the ipsec configuration files.  The default is operating system specific and you should not need to override this setting.
+# @param ipsec_conf
+#    The location of the `ipsec.conf` file. The default is operating system specific and you should not need to override this setting.
+# @param ipsec_secrets
+#    The location of the `ipsec.secrets` file. The default is operating system specific and you should not need to override this setting.
+# @param charon_conf
+#    The location of the main `charon.conf` config file. WARNING, this option is not currently used.
 class strongswan(
-  $package_name        = $strongswan::params::package,
+  $package_name        = 'strongswan',
```

Comment:
Non operating specific defaults are better in init.pp directly where puppet-strings can find them.